File: darkwater-api/src/main/java/kiinse/plugins/darkwaterapi/api/commands/CommandContext.java
package kiinse.plugins.darkwaterapi.api.commands;

import kiinse.plugins.darkwaterapi.api.files.locale.PlayerLocale;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

public interface CommandContext {

    @NotNull CommandSender getSender();

    @NotNull String[] getArgs();

    @NotNull PlayerLocale getSenderLocale();

}

File: api/src/main/kotlin/kiinse/me/plugins/darkwaterapi/api/commands/CommandContext.kt
package kiinse.me.plugins.darkwaterapi.api.commands

import kiinse.me.plugins.darkwaterapi.api.files.locale.PlayerLocale
import org.bukkit.command.CommandSender

@Suppress("UNUSED")
interface CommandContext {
    val sender: CommandSender
    val args: Array<String>
    val senderLocale: PlayerLocale
}